session/auth/authprovider.go/authprovider.go

```diff
@@ -0,0 +1,44 @@
+package authprovider
```

Comment:
why the package directory is named `authprovider.go`?

Reply:
will fix

Reply:
updated